### DISCUSSION TITLE
Pixi.Text() only as Texture-provider?

### DISCUSSION THREAD

**Discussion by joergplewe** [pixijs/pixijs#8016] [Q&A]
Hi all!

I have to deal with lots of non-changing text objects. My idea is to use `Pixi.Text` as a provider for a texture and create a `Pixi.Sprite` from it. Like so:

```
const txt = new Text(text, style);
txt.updateText(false);
container.addChild(new Sprite(txt.texture));
```
This actually works but leaves me with dangling `Pixi.Text` objects, each allocating a canvas and such, eating up memory.

So I added a `destroy()`:

```
const txt = new Text(text, style);
txt.updateText(false);
container.addChild(new Sprite(txt.texture));
txt.destroy({texture: false});
```
which gives me a 

`13WebGL: INVALID_VALUE: texImage2D: no canvas`

Next, I tried to clone the texture before destroying the `Pixi.Text`:

```
const txt = new Text(text, style);
txt.updateText(false);
const texture = txt.texture.clone();
container.addChild(new Sprite(txt.texture));
txt.destroy();
```
which gives me just empty images.

So ... why doesn't the texture cloning work? Are there better ways to deal with that?

**Comment by bigtimebuddy** [comment]
Couple of things:

First your assumption about using Text to create a Sprite. That is, in fact, how Text works. It's a Sprite with some stuff for redrawing a canvas under the hood. It's better, and less complex, to use Text but _don't_ constantly update the text content or style. Any time you redraw it requires another GPU upload, which is usually the bottleneck.

As for your second question, Text manages the Texture, BaseTexture and canvas. When you destroy it destroys all of these things. Text's textures aren't really meant to be reused like this after destroying Text.

Hope that helps clarify.

**Reply by joergplewe** [reply]
Hi and thanks. 

Just another question after looking at the source `Text.ts`.

```
public destroy(options?: IDestroyOptions|boolean): void
    {
        if (typeof options === 'boolean')
        {
            options = { children: options };
        }

        options = Object.assign({}, defaultDestroyOptions, options);

        super.destroy(options);
```

It seems that `txt.destroy({children: true, texture: false, baseTexture: false});` should prevent `Text` from deleting the texture? Which brings me to the question of how canvas and texture are related (bc. it is set to size 0)?

Greets,

. J

**Comment by ivanpopelyshev** [comment]
clone() clones only rectangle coords, not the texture data. Just remove `txt.destroy()` and it should work fine.

**Reply by joergplewe** [reply]
Hi!

Is there a way to clone a texture with content?

I'm a bit concerned that added-nowhere `Text` objects could be dangling somewhere causing a memory leak?

**Reply by ivanpopelyshev** [reply]
baseTexture is the only thing that can dangle and you're gonna use it. Rest of text will go away just fine. As for cloning, somewhere else has to answer you, i wrote that thing 100 times and im lazy.
